File: backend/Models/Company.cs
namespace InternshipManagement.Api.Models;

public class Company
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string TaxCode { get; set; } = string.Empty;

    public string? ContactEmail { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<User> Users { get; set; }
        = new List<User>();

    public ICollection<Application> Applications { get; set; }
        = new List<Application>();
}

File: frontend/src/types/company.ts
export interface CompanyResponse {
  id: string;
  name: string;
  taxCode: string;
  contactEmail: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCompanyRequest {
  name: string;
  taxCode: string;
  contactEmail: string | null;
}

export interface UpdateCompanyRequest {
  name: string;
  taxCode: string;
  contactEmail: string | null;
  isActive: boolean;
}